src/actions/VaccineActions.js

```diff
@@ -170,32 +190,44 @@ const startDownloadAllLogs = macAddress => async (dispatch, getState) => {
 };
 
 const setLogInterval = (macAddress, interval) => async dispatch => {
-  let ok = false;
-  let error = `Sensor response was not equal to 'Interval: ${interval}s'`;
-
   dispatch(setLogIntervalStart(macAddress));
+  const regex = new RegExp(`Interval: ${interval}s`); // TODO: update with sensor specific response as needed
+  const error = `Sensor response was not equal to 'Interval: ${interval}s'`;
 
-  try {
-    const result = await BleService().updateLogIntervalWithRetries(macAddress, interval, 3, error);
-    const regex = new RegExp(`Interval: ${interval}s`); // TODO: update with sensor specific response as needed
-    ok = regex.test(result);
-  } catch (e) {
-    error = e;
-    ok = false;
-  }
-
-  if (ok) {
-    dispatch(setLogIntervalSuccess());
-    ToastAndroid.show(
-      vaccineStrings.formatString(vaccineStrings.log_interval_set, interval),
-      ToastAndroid.LONG
-    );
-  } else {
-    ToastAndroid.show(vaccineStrings.E_COMMAND_FAILED, ToastAndroid.LONG);
-    dispatch(setLogIntervalError(error));
-  }
+  return BleService()
+    .updateLogInterval(macAddress, interval)
+    .then(result => (regex.test(result) ? setLogIntervalSuccess() : setLogIntervalError(error)))
+    .then(action => dispatch(action));
+};
 
-  return ok;
+const saveSensor = sensor => async dispatch =>
+  new Promise((resolve, reject) => {
+    dispatch(saveSensorStart(sensor.macAddress));
+
+    try {
+      const { location, logInterval, macAddress, name } = sensor;
+      const sensorManager = SensorManager();
+      const newSensor = sensorManager.createSensor({ location, logInterval, macAddress, name });
+      sensorManager.saveSensor(newSensor);
+      dispatch(saveSensorSuccess());
+      resolve();
+    } catch (error) {
+      dispatch(saveSensorError());
+      reject(error);
```

Comment:
If you just throw this, then I'm not sure you need to the `new Promise`?

Reply:
I'm using the `new Promise` because `saveSensor` is 'promise-like' only ( aka 'thenable' ) and supports `then` but not `catch`

Reply:
Just been running this but I can't actually get a sensor to complete the steps before it's disconnected.

But I'm very confused by this statement? `saveSensor` returns a promise, no?

Reply:
Oh - I was thinking you meant `sensorManager.saveSensor()` but you mean this function?